chain-signatures/node/src/protocol/signature.rs

```diff
@@ -680,7 +675,7 @@ impl SignatureManager {
             }
 
             let Some(my_request) = my_requests.pop_front() else {
-                failed_presigs.push(presignature);
+                // TODO: we are loosing presignature now with no good reason
```

Comment:
@ChaoticTempest I'm not fixing this TODO, since this part if rewritten in your PR.